srtcore/core.cpp

```diff
@@ -677,6 +686,10 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
 
             m_iSndCryptoKeyLen = v;
         }
+#else
+        LOGC(mglog.Error, log << "SRTO_PASSPHRASE: encryption not enabled at compile time");
+        throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
+#endif
```

Comment:
Maybe just preserve the existing function. It will simply return zero.

Reply:
Setting an encryption-related option is generally wrong in case when you have a version with encryption not enabled at compile time. I'd personally cut off much more from the functionality (even to the grade that makes both these options not even recognized as valid options), but this would cost too much effort with very little profit.